tutorial/lesson_01_basics.cpp

```diff
@@ -3,11 +3,11 @@
 // This lesson demonstrates basic usage of Halide as a JIT compiler for imaging.
 
 // On linux, you can compile and run it like so:
-// g++ lesson_01*.cpp -g -I <path/to/Halide.h> -L <path/to/libHalide.so> -lHalide -lpthread -ldl -o lesson_01 -std=c++11
+// g++ lesson_01*.cpp -g -I <path/to/Halide.h> -L <path/to/libHalide.so> -lHalide -lpthread -ldl -o lesson_01 -std=c++17
```

Comment:
Since "using" Halide doesn't require c++17 -- do we need to change the compile flags in the tutorials? I don't have strong opinions, just asking.

Reply:
See the top comment above. Probably should keep some/all at C++11 (I just changed everything in a frenzy before thinking about it), but didn't go back and undo those changes yet to see what folks thought -- in particular, are we ready to declare it requiring C++17 for usage too? (Maybe not, but it would certainly be fine at Google.)